### DISCUSSION TITLE
wrong coordinates for predictions

### DISCUSSION THREAD

**Discussion by AdwaitMD** [ultralytics/ultralytics#13410] [Q&A]
Hi, 
I have trainesdyolov8-pose. I get good results for the prediction of multiple skeletons and bounding boxes.. however, for many frames.. some skeletons (all key points) have wrong coordinates. It looks like x and y are flipped.. or something like that is happening.  or maybe the origin (0,0) is not right for these skeletons. Because the poses look correct but just not on the object.. they are located where the mirror image of that object would be located.  I would appreciate any solutions for this.   my model was trained on imgsz =1280; I am putting a 5k image in the prediction, is this has something to do with wrong coorinates.  Is it not scaling back well.  Thanks.

**Comment by glenn-jocher** [comment]
@AdwaitMD hi,

It sounds like the issue might be related to how the image resizing or coordinate scaling is handled during prediction, especially since you mentioned using a different image size (5k) than what the model was trained on (1280). Here are a few steps you can try:

1. **Consistent Image Size**: Ensure that the image size during prediction matches the `imgsz` used during training. If resizing is necessary, maintain the aspect ratio or use padding to match the training dimensions.

2. **Coordinate Check**: Verify if the coordinate flipping issue persists when using images of size 1280 during prediction. This can help isolate whether the problem is due to resizing or a model training issue.

3. **Post-Processing**: Check any post-processing steps where coordinates are transformed or scaled. Ensure transformations align with your model's training setup.

Adjusting these factors should help correct the coordinate discrepancies you're observing. If the issue persists, consider re-evaluating the data preprocessing and augmentation steps used during training.

**Reply by AdwaitMD** [reply]
Hi, 
I have tried everything you have suggested. Now, it seems like it's not even mirroring. Some key points are outside the bounding boxes. Sometimes even, there are extra key points (skeletons) than bounding boxes. This is happening to a minority of detections. Others are perfect. 
What else I Can try? Thanks.

**Reply by glenn-jocher** [reply]
Hi @AdwaitMD,

Thanks for the update. Given the inconsistencies, I recommend ensuring you're using the latest version of the Ultralytics package. If the issue persists, try validating the model on a smaller, controlled dataset to isolate the problem. Additionally, double-check your data augmentation and preprocessing steps to ensure they align with the model's expectations. If the problem continues, please share more details about your setup and any specific patterns you've noticed.

**Reply by AdwaitMD** [reply]
HI @glenn-jocher , 

Thanks for the help. Miraculously, I found the solution. By removing "augment =True" from the model, predict(), the wrong, mirrored, and extra keypoint skeletons disappeared from the Inference. I checked with different trained models and also different datasets for inference. It may be useful information for you for further development.

**Reply by glenn-jocher** [reply]
Hi @AdwaitMD,

Great to hear you found a solution! Removing `augment=True` can indeed affect the prediction results. Thanks for sharing this insight; it could be valuable for others facing similar issues. If you encounter any further problems or have more questions, feel free to reach out.
